File: python/pali.py
def isPalindrome(arr_or_str):
    i = 0
    j = len(arr_or_str) - 1

    while i <= j:
        if arr_or_str[i] != arr_or_str[j]:
            return False
        i += 1
        j -= 1

    return True

File: java/algods-java/src/main/java/com/example/langaaegeskov/Pali.java
package com.example.langaaegeskov;

public class Pali {
    public static void main(String str)
    {
        System.out.println( isPalindrome(str));
        
    }

    public static boolean isPalindrome (String str) {
        int i = 0;
        int j = str.length() - 1;
         
        while (i <= j) {
            if (str.charAt(i) != str.charAt(j)) {
                return false;
            }
            ++i;
            --j;
        }

        return true;
    }
}
